client/dashboard/components/segmented-bar/index.tsx

```diff
@@ -0,0 +1,160 @@
+	/** Numeric value that determines the segment size */
+	value: number;
+	/** Optional text label used for accessibility and analytics */
+	label?: string;
+	/** Optional custom color; falls back to palette in CSS */
+	color?: string;
+};
+
+export type SegmentedBarProps = {
+	/**
+	 * Array of segments to render. The relative size of each segment is
+	 * determined by its value relative to the sum (or `total` if provided).
+	 */
+	segments: SegmentedBarSegment[];
+	/**
+	 * When provided, percentages are computed against this number instead of
+	 * the sum of segment values. Useful if you need to normalize against a
+	 * known total that includes hidden buckets.
+	 */
+	total?: number;
+	/**
+	 * Minimum percentage shown for a non-zero segment to keep it visible.
+	 * Defaults to 2.5% to match small UI density.
+	 */
+	minPercent?: number;
+	/** Optional overall ARIA label for the segmented bar */
+	ariaLabel?: string;
+	/** Height in pixels, defaults to 16 */
+	height?: number;
+	/** Border radius in pixels, defaults to 0 (square ends) */
+	radius?: number;
+	/** Optional gap between segments in pixels. Defaults to 6 (some spacing). */
+	gap?: number;
+	/** Render labels row above the bar. Defaults to true. */
+	showLabels?: boolean;
+	/** Custom renderer for label values (e.g., currency). Defaults to raw value. */
+	formatValue?: ( value: number, segment: SegmentedBarSegment ) => React.ReactNode;
+};
+
+/**
+ * Converts raw numeric values into normalized percentages while ensuring a
+ * minimum visible size for non-zero buckets.
+ */
+function normalizePercents( values: number[], minPercent: number ): number[] {
+	const sum = values.reduce( ( acc, v ) => acc + v, 0 );
+	if ( sum <= 0 || values.length === 0 ) {
+		// Nothing to show; render empty bar
+		return values.map( () => 0 );
+	}
+
+	// Raw percentages based on provided values
+	const raw = values.map( ( v ) => ( v <= 0 ? 0 : ( v / sum ) * 100 ) );
+
+	// Ensure visibility for tiny, non-zero buckets
+	const adjusted = raw.map( ( p, idx ) => ( values[ idx ] > 0 ? Math.max( p, minPercent ) : 0 ) );
+	const adjustedTotal = adjusted.reduce( ( a, b ) => a + b, 0 ) || 1;
+
+	// Normalize back to exactly 100%
+	return adjusted.map( ( p ) => ( p / adjustedTotal ) * 100 );
+}
+
+/**
+ * SegmentedBar renders a single horizontal bar divided into proportional
+ * segments. It is intentionally minimal and CSS-driven so it can be moved
+ * to a shared chart package in the future.
+ */
+export default function SegmentedBar( {
+	segments,
+	total,
+	minPercent = 2.5,
+	ariaLabel,
+	height = 16,
+	radius = 0,
+	gap = 6,
+	showLabels = true,
+	formatValue,
+}: SegmentedBarProps ) {
+	const isSmallViewport = useViewportMatch( 'medium', '<' );
+	const values = segments.map( ( s ) => s.value );
+	const effectiveSum = typeof total === 'number' ? total : values.reduce( ( a, b ) => a + b, 0 );
+
+	const percents =
+		effectiveSum > 0 ? normalizePercents( values, minPercent ) : values.map( () => 0 );
```

Comment:
I may be misunderstanding the code, but based on the comment for the `total` prop, when passed to the component, it should be used to normalize the percentages as opposed to using the total sum. I don't see the `total` being passed in here.

Reply:
No problem 👍